arviz/data/inference_data.py

```diff
@@ -23,8 +23,13 @@
     Union,
     overload,
 )
+try:
+    import h5netcdf.legacyapi as nc
+    HAS_H5NETCDF = True
+except ImportError:
+    import netCDF4 as nc
+    HAS_H5NETCDF = False
```

Comment:
I'm thinking of this, and given that we require this, then we don't need to have this fail-safe check here.

So I think we can just remove this check and also use the main API?

Reply:
I certainly don't have any objections! The docs for h5netcdf suggest that it should be a drop-in replacement for netCDF4 except in some edge cases that I don't think get used here.